v2/fb-xcom-api-validator/src/signing.ts

```diff
@@ -0,0 +1,129 @@
+import { createHmac, createPrivateKey, createPublicKey, createSign, createVerify } from "crypto";
+
+export class InvalidSignatureError extends Error {
+    public get responseObject() {
+        return {
+            errorCode: "invalid-signature",
+        }
+    }
+}
+export class AlgorithmNotSupportedError extends Error {}
+
+type HashingAlgorithm = "sha256" | "sha512" | "sha3-256"
+type SigningAlgorithm = "hmac" | "rsa" | "ecdsa"
+type NamedCurve = "prime256v1" | "secp256k1"
+type Options = { algorithm: HashingAlgorithm, curve?: NamedCurve }
```

Comment:
`algorithm` here is very misleading. `hashAlgorithm`?

Reply:
Agreed, as suggested, now using `hashAlgorithm`